jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -66,25 +67,35 @@ class CodeWhispererRecommendationManager {
 
         val reformattedRecommendation = tempDocument.getText(TextRange(tempRangeMarker.startOffset, tempRangeMarker.endOffset))
 
-        // Build new reference with updated contentSpan(start and end). Since it's reformatted, take the new
-        // start and end from the rangeMarker which automatically tracks the range after reformatting
-        val reformattedReferences = rangeMarkers.map {
-            val currentRange = it.key
-            val originalReference = it.value
-            originalReference
+        val reformattedReferences = rangeMarkers.map { (rangeMarker, reference) ->
+            reformatReference(reference, rangeMarker, invocationStartOffset)
+        }
+        return Recommendation.builder()
+            .content(reformattedRecommendation)
+            .references(reformattedReferences)
+            .build()
+    }
+
+    /**
+     * Build new reference with updated contentSpan(start and end). Since it's reformatted, take the new start and
+     * end from the rangeMarker which automatically tracks the range after reformatting
+     */
+    fun reformatReference(originalReference: Reference, rangeMarker: RangeMarker, invocationStartOffset: Int): Reference {
+        rangeMarker.apply {
+            val documentContent = runReadAction { document.charsSequence }
```

Comment:
not sure if we need this runReadAction -- considering we are inside edt?

Reply:
let me try this